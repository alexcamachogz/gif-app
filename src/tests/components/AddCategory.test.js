import React from 'react'
import { shallow } from 'enzyme'
import AddCategory from '../../components/AddCategory'

describe('Testing AddCategory component', () => {

  const setCategories = jest.fn()
  let wrapper
  beforeEach(() => {
    jest.clearAllMocks()
    wrapper = shallow(<AddCategory setCategories={setCategories}/>)
  })

  test('Should be show <AddCategory /> properly', () => {
    expect(wrapper).toMatchSnapshot()
  })

  test('Input text should be change', () => {
    const input = wrapper.find('input')
    const value = 'Hello World!'
    input.simulate('change', { target: { value } })
    expect(input.prop('value')).toBe('')
  })

  test('Not should post the information with submit...', () => {
    wrapper.find('form').simulate('change', { preventDefault () {} })
    expect(setCategories).not.toHaveBeenCalled()
  })

  test('Should call setCategories and clean input text', () => {
    /* Tarea:
    * Simular el input change
    * Simular el submit
    * Llamar setCategories
    * inputValue debe de estar ''
    * */

    const value = 'Hello World!'

    wrapper.find('input').simulate('change', { target: { value } })
    wrapper.find('form').simulate('submit', { preventDefault () {} })
    expect(setCategories).toHaveBeenCalled()
    expect(wrapper.find('input').prop('value')).toBe('')
  })

})