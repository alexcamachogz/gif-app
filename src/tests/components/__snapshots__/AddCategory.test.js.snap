// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing AddCategory component Should be show <AddCategory /> properly 1`] = `
<div
  className="container"
>
  <form
    onSubmit={[Function]}
  >
    <input
      onChange={[Function]}
      placeholder="Buscar por palabra clave"
      type="text"
      value=""
    />
  </form>
</div>
`;
